solutions/usaco-991.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: usaco-991
+source: USACO Silver 2020 January
+title: Loan Repayment
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_loan_silver_jan20.html)
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(\sqrt{N} \cdot \log{N})$
+
+<LanguageSection>
+  
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class LoanRepayment {
+	// our checking function in the binary search
+	static boolean isOk (long numGallons, long withinDays, long atLeast, long x) {
+		long g = 0;
+		while (withinDays > 0 && g < numGallons) {
+			long y = (numGallons - g) / x;
```

Comment:
explain some of the math here?

Reply:
it is in the problem statement though

" 1. Assuming that Farmer John has already given Bessie G gallons, compute N−GX rounded down. Call this number Y.

   2. If Y is less than M, set Y to M.
   
   3. Give Bessie Y gallons of milk. "

Reply:
yeah that's kinda bad, just put some simple explanation
even just "compute amount of milk to give" would be nice